### PR TITLE
minor code revision

### PR DESCRIPTION
**Pull request by thoeschler**

### CODE REVIEW

File: Solver/change_dimensions.py [hunk 1]
```diff
@@ -85,111 +86,110 @@ def assign_bc_type(bc):
...
             material_parameters_dimless, u_ana_dimless
-    else: # no analytical solution
+    else:  # no analytical solution
         return dom_dimless, boundary_conditions_dimless, body_forces_dimless, \
             material_parameters_dimless
 
 
-def get_dimensionful_values(u_hat_dimless, boundary_conditions, nondim_disp, nondim_length):
+def get_dimensionful_values(u_hat_dimless, boundary_conditions,
+                            nondim_disp, nondim_length):
```
Comment (sebglane) [inline Solver/change_dimensions.py:192]: Rename to `get_dimensional_values`.

File: Solver/change_dimensions.py [hunk 2]
```diff
@@ -211,34 +211,42 @@ def get_dimensionful_values(u_hat_dimless, boundary_conditions, nondim_disp, non
...
-    dom_dimless = tuple([u_hat_dimless.function_space().spaces[i].bases[i].domain for i in range(dim)])
+    dom_dimless = [u_hat_dimless.function_space().spaces[i].bases[i].domain
+                   for i in range(dim)]
 
-    # dimensionful domain  
-    dom = tuple([tuple([dom_dimless[i][j]*nondim_length for j in range(2)]) for i in range(dim)])
+    # dimensionful domain
+    dom = [
+            [
+                dom_dimless[i][j]*nondim_length for j in range(2)
```
Comment (sebglane) [inline Solver/change_dimensions.py:228]: Why is there `for j in range(2)` and not `for j in range(dim)`?

File: Solver/check_solution.py [hunk 3]
```diff
@@ -1,10 +1,12 @@
 from shenfun import div, grad, Function, project, Array, inner
 from math import sqrt
 
+
 def check_solution_cauchy(u_hat, material_parameters, body_forces):
     '''
-    Check whether the numerically computed solution fulfills the Lamé-Navier equation.
-    
+    Check whether the numerically computed solution fulfills the
+    Lamé-Navier equation.
```
Comment (sebglane) [inline Solver/check_solution.py:8]: Do not use non-ASCII characters such as é. Change to Lame-Navier equation**s**.

File: Solver/check_solution.py [hunk 4]
```diff
@@ -26,34 +29,33 @@ def check_solution_cauchy(u_hat, material_parameters, body_forces):
     assert len(material_parameters) == 2
     for val in material_parameters:
         assert isinstance(val, float)
-    
+
     # some parameters
-    lambd = material_parameters[0]
-    mu = material_parameters[1]
-    
+    lambd, mu = material_parameters
```
Comment (sebglane) [inline Solver/check_solution.py:34]: I prefer lmbda.

File: Solver/change_dimensions.py [hunk 5]
```diff
@@ -211,34 +211,42 @@ def get_dimensionful_values(u_hat_dimless, boundary_conditions, nondim_disp, non
...
     u_hat = Function(V)
-        
+
     for i in range(dim):
-        u_hat[i] = nondim_disp*u_hat_dimless[i] # u has the same expansions coefficients as u_hat
-            
-    return u_hat
\ No newline at end of file
+        # u has the same expansions coefficients as u_hat
+        u_hat[i] = nondim_disp*u_hat_dimless[i]
```
Comment (sebglane) [inline Solver/change_dimensions.py:250]: Operator spacing!

File: Solver/change_dimensions.py (hunk 2 above)
Comment (sebglane) [inline Solver/change_dimensions.py:228]: Operator spacing!

File: Solver/body_forces.py [hunk 6]
```diff
@@ -71,63 +68,55 @@ def body_forces_gradient(u, material_parameters):
...
         GradDivDivGrad[i] = DivDivGrad.diff(coord[i])
-    
+
     # body forces
-    body_forces = tuple([
-            (c1 + c4)*DoubleLaplace[i] + (c2 + c3 + c5)*GradDivDivGrad[i] \
-            - (lambd + mu)*GradDiv[i] - mu*Laplace[i] for i in range(dim)
-        ])
+    body_forces = (c1 + c4)*DoubleLaplace + (c2 + c3 + c5)*GradDivDivGrad \
+        - (lambd + mu)*GradDiv - mu*Laplace
```
Comment (sebglane) [inline Solver/body_forces.py]: Operator spacing for * !

File: Solver/body_forces.py [hunk 7]
```diff
@@ -14,52 +15,48 @@ def body_forces_cauchy(u, material_parameters):
...
         for j in range(dim):
             Laplace[i] += u[i].diff(coord[j], 2)
-            
+
     # compute body forces
-    body_forces = tuple([
-           - ( (lambd + mu)*GradDiv[i] + mu*Laplace[i] ) for i in range(dim)
-        ])
-        
+    body_forces = - (lambd + mu)*GradDiv - mu*Laplace
```
Comment (sebglane) [inline Solver/body_forces.py]: Operator spacing for  * !

File: Solver/stresses.py [hunk 8]
```diff
@@ -20,43 +21,30 @@ def cauchy_stresses(material_parameters, u_hat):
...
+
     # Cauchy stress tensor
-    T = [ [None for _ in range(dim)] for _ in range(dim)]
-    for i in range(dim):
-        for j in range(dim):
-            T[i][j] = 2.0 * mu * E[i][j] 
-            if i==j:
-                T[i][j] += lambd * trE
-            
+    T = 2.0 * mu * E + lambd * trE * np.identity(dim)
```
Comment (sebglane) [inline Solver/stresses.py]: That is great!

File: Solver/stresses.py [hunk 9]
```diff
@@ -79,34 +67,28 @@ def hyper_stresses(material_parameters, u_hat):
...
-    GradDiv = [0. for _ in range(dim)]
+    GradDiv = np.empty(dim)
     for i in range(dim):
-        for j in range(dim):
-            GradDiv[i] += project(Dx(Dx(u_hat[j], j), i), T_none)
-    
+        GradDiv[i] += project(Dx(Div, i), T_none)
+
     # hyper stresses
     T = [ [ [0. for _ in range(dim)] for _ in range(dim)] for _ in range(dim)]
```
Comment (sebglane) [inline Solver/stresses.py]: May you could use a `np.empty` object here too...

**Review by sebglane — CHANGES_REQUESTED** [review]
Good to go if you implement my proposed changes.

File: Solver/change_dimensions.py (hunk 2 above)
Comment (thoeschler) [reply Solver/change_dimensions.py:228]: This refers to the start and end point of the interval so it is always equal to 2.

**Review by sebglane — APPROVED** [review]

**Pull request merged**